.github/workflows/test-and-publish-nightly.yml

```diff
@@ -28,51 +25,46 @@ jobs:
   build-binaries-and-test:
     strategy:
       matrix:
-        runson:
-          - display: linux-x64
-            name: ubuntu-latest # Using "latest" here as the build and test will any ways run inside a container which we control
+        name: [ubuntu-latest] # Using "latest" here as the build and test will any ways run inside a container which we control
         java-version: [11, 17, 21, 24]
         java-distribution: [corretto]
+        architecture: [x64]
         container: ["public.ecr.aws/async-profiler/asprof-builder-x86:latest"]
         include:
-          - runson:
-              display: macos-14-arm64
-              name: macos-14
+          - name: macos-13
             java-version: 11
             java-distribution: corretto
+            architecture: x64
+            container: ""  # Not using container for mac-os as we have images only for linux
+          - name: macos-14
+            java-version: 21
+            java-distribution: corretto
+            architecture: arm64
             container: ""  # Not using container for mac-os as we have images only for linux
-          - runson:
-              display: linux-arm64
-              name: ubuntu-24.04-arm # There is no "latest" tag available (yet) as ARM runners are still in public preview
+          - name: ubuntu-24.04-arm # There is no "latest" tag available (yet) as ARM runners are still in public preview
             java-version: 11
             java-distribution: corretto
+            architecture: arm64
             container: "public.ecr.aws/async-profiler/asprof-builder-arm:latest"
-    runs-on: ${{ matrix.runson.name }}
+    runs-on: ${{ matrix.name }}
     container:
       image: ${{ matrix.container }}
       options: --privileged
-    name: "test (${{ matrix.runson.display }}, ${{ matrix.java-distribution }} ${{ matrix.java-version }})"
+    name: "test (${{ matrix.name }}, ${{ matrix.java-distribution }} ${{ matrix.java-version }}${{ matrix.architecture && ' ' || '' }}${{ matrix.architecture }}"
```

Comment:
`architecture` is now always defined, right?

Reply:
Yes, should I enforce it with a check?